### DISCUSSION TITLE
PR Backlog: Highly Requested PRs, Ready For Review, and more...

### DISCUSSION THREAD

**Discussion by Swivelgames** [polybar/polybar#3110] [Q&A]
As a one-man-show, I imagine @patrick96 has some difficulty keeping up with all of the activity in this repo. Many of the newest Issues and Pull Requests have had a tendency to cloud the important PRs that have been hanging out in the backlog for quite some time.

In the hopes of lending a hand, I went ahead and created a categorized list PRs.

Some of these PRs are 7 years old, but are still actively used by community members maintaining Frankenstein forks, that ultimately fragment the community around `polybar`.

I've categorized each of the PRs with the following labels:

| Label | Assignee | Description |
| :---- | :-- | :---------- |
| :fire: **Hot** |  | These PRs are highly requested by the community. |
| :white_check_mark: **Approved, Never Merged** | polybar | PRs that have been approved, but were never released. |
| :eyes: **Ready For Review** | polybar | Awaiting Review or Re-Review |
| :new: **New PRs** | polybar | Fresh PRs, ready to be reviewed |
| :construction: **Changes Needed** | Author | Changes were requested by the Polybar team. The owner of this PR should update it so that it can be reviewed again. |
| :sos: **Help-Wanted** | Author | These PRs need some love before they can be reviewed. Usually, these were close to being approved, but needed changes. Often times these need to be rebased due to their age. |
| :hourglass: **Stale** | Author | These PRs were either rejected or abandoned, and can probably be safely closed. They can always be reopened if they're still wanted or needed! |

Stale PRs can probably be closed, and if they're still needed, can always be reopened later :+1: 

The hotly-anticipated PRs are ones that have been seemingly approved, ready to be merged, and/or have a decent level of community activity in them hoping for a merge.

# Hot PRs Summary

Listed in chronological order.

- :fire: :sos: **Help-Wanted** | #1075
- :fire: :white_check_mark: **Approved, Never Merged** | #1478
- :fire: :white_check_mark: **Approved, Never Merged** | #1590 (milestone 3.8.0 :tada: )
- :fire: :white_check_mark: **Approved, Never Merged** | #1668
- :fire: :sos: **Help-Wanted** | #2151
- :fire: :construction: **Changes Needed** | #2486
- :fire: :eyes: **Ready For Review** | #2868
- :fire: :sos: **Help-Wanted** | #2959
- :fire: :eyes: **Ready For Review** | #2980
- :fire: :eyes: **Ready For Review** | #3006

# 6-7 Years Old (2017 + 2018)

Many of these PRs were contributions by the late @Lomadriel, who departed Github nearly 3 years ago on Aug 6, 2020 :bow:

- :fire: **Hot** | #1075
- :fire: **Hot** | #1478
- :white_check_mark: **Approved, Never Merged** | #1306
- :white_check_mark: **Approved, Never Merged** | #1515
> - :sos: **Help-Wanted** | #1122
> - :hourglass: **Stale** | #725 
> - :hourglass: **Stale** | #1286
> - :hourglass: **Stale** | #1409
> - :hourglass: **Stale** | #1538

</details>

# 5 Years Old (2019)

- :fire: **Hot** | #1590 (milestone 3.8.0 :tada: )
- :fire: **Hot** | #1668
- :white_check_mark: **Approved, Never Merged** | #1673
- :white_check_mark: **Approved, Never Merged** | #1790
- :eyes: **Ready For Review** | #1947
> - :sos: **Help-Wanted** | #1574
> - :sos: **Help-Wanted** | #1684
> - :sos: **Help-Wanted** | #1686
> - :sos: **Help-Wanted** | #1704
> - :sos: **Help-Wanted** | #1768
> - :sos: **Help-Wanted** | #1809
> - :sos: **Help-Wanted** | #1810
> - :sos: **Help-Wanted** | #1963
> - :hourglass: **Stale** | #1687
> - :hourglass: **Stale** | #1696
> - :hourglass: **Stale** | #1779
> - :hourglass: **Stale** | #1852
> - :hourglass: **Stale** | #1938 

# 4 Years Old (2020)

- :fire: **Hot** | #2151
- :white_check_mark: **Approved, Never Merged** | #2038
- :eyes: **Ready For Review** | #1998
- :eyes: **Ready For Review** | #2013
- :eyes: **Ready For Review** | #2065
- :eyes: **Ready For Review** | #2076
- :eyes: **Ready For Review** | #2104
- :eyes: **Ready For Review** | #2109
- :eyes: **Ready For Review** | #2220
- :eyes: **Ready For Review** | #2238 (related: #2486)
> - :sos: **Help-Wanted** | #2037
> - :hourglass: **Stale** | #2157

# 3 Years Old (2021)

- :fire: **Hot** | #2486
- :sos: **Help-Wanted** | #2454
- :sos: **Help-Wanted** | #2515
- :hourglass: **Stale** | #2386

# 2 Years Old (2022)

- :fire: **Hot** | #2868
- :eyes: **Ready For Review** | #2645
- :eyes: **Ready For Review** | #2714
- :eyes: **Ready For Review** | #2735
- :eyes: **Ready For Review** | #2856

> - :construction: **Changes Needed** | #2797
> - :construction: **Changes Needed** | #2853
> - :construction: **Changes Needed** | #2858
> - :construction: **Changes Needed** | #2881
> - :hourglass: **Stale** | #2748

# 2023 + 2024 :tada: 

- :fire: **Hot** | #2959
- :fire: **Hot** | #2980
- :fire: **Hot** | #3006
- :eyes: **Ready For Review** | #3033
- :eyes: **Ready For Review** | #3078
- :eyes: **Ready For Review** | #3096
---
- :new: **New PRs** | #2898
- :new: **New PRs** | #2976
- :new: **New PRs** | #2985
- :new: **New PRs** | #2999
- :new: **New PRs** | #3001
- :new: **New PRs** | #3093
- :new: **New PRs** | #3095
- :new: **New PRs** | #3103
- :new: **New PRs** | #3104
- :new: **New PRs** | #3105
- :new: **New PRs** | #3108

**Comment by patrick96** [comment]
Thank you putting together this summary, I do really appreciate it (and sorry for ignoring your email).
Having two pages of pending github notifications can be quite overwhelming, so this should definitely help make this more manageable.

I have been struggling to keep up with any maintenance work in polybar.
Since I started working full-time, coding in my free time just isn't as fun anymore when I'm already doing it all day at work.
I also just have much less spare time on my hands, these days I'm only booting up my personal machine about once a week.

So, expect progress here to keep being pretty slow in the future and only more critical or sought-after things to get attention.
But I still have big plans for the project, they're just going to be happening at a slower pace than I'd like.

**Reply by Swivelgames** [reply]
I totally get it! I'm in a very similar position. First and foremost, I rarely ever check my personal email unless I'm grabbing a OTP code :joy: It can feel like a runaway train with two Issues or PRs being added to the project every time you finally close one of them. I hope things smooth out for you and projects like `polybar` become fun again. At least for me, it's an ebb and flow. There are some times I go weeks without turning on my personal machine, and other times that I spent most evenings working on side projects.

I think most all of us as developers are in a similar boat, so I wanted to try something helpful as a way to give back to a great project, and to show my appreciation. Hopefully the lists I produced above make it easier (and not harder) to go through and close some old PRs, prioritize others, and cut the number of PRs and Issues down by quite a bit so that it isn't as overwhelming. :slightly_smiling_face: :+1:
